library/Director/Resolver/CommandUsage.php

```diff
@@ -89,14 +94,26 @@ protected function fetchFor($table, $rels, $objectTypes)
     {
         $id = $this->command->getAutoincId();
 
+        //return (object) array('object'=>1);
         $columns = [];
         foreach ($objectTypes as $type) {
             $columns[$type] = "COALESCE(SUM(CASE WHEN object_type = '$type' THEN 1 ELSE 0 END), 0)";
         }
-        $query = $this->db->select()->from("icinga_$table", $columns);
 
-        foreach ($rels as $rel) {
-            $query->orWhere("{$rel}_id = ?", $id);
+        if ($table === "command") {
+            $query = $this->db->select()->from(array('c' => 'icinga_' . $table), $columns)->join(array('ici' => 'icinga_command_inheritance'),
+                'ici.command_id = c.id',
+                array());
+
+            foreach ($rels as $rel) {
+                $query->orWhere("parent_{$rel}_id = ?", $id);
+            }
```

Comment:
```suggestion
            $query = $this->db->select()->from(['c' => 'icinga_' . $table], $columns)->join(
                ['ici' => 'icinga_command_inheritance'],
                'ici.command_id = c.id',
                []
            )->where('ici.parent_command_id = ?', $id);
```

IMO, you could rewrite the `where` as suggested, just makes it a bit easier to understand the code

Reply:
this would render the $rels var useless, and might break something else

Reply:
The `$rels` variable comes from the `$map` variable. You could give an empty array there, as the relation in the map is basically used to build the column `{rel}_id` to filter the objects. And since for `command` you filter  its inheritance table. You could directly filter it in the query.

Or you could do the below, where you add the join and the filter to the query only if the table is 'command', and everything remains same. This solution also results in minimal changes to the existing code. You should still give empty array for command (`'command' => []`) in `$map` variable. Also, looking at it again I find this solution better.

```php
        $query = $this->db->select()->from(['c' => "icinga_$table"], $columns);
        if ($table === "command") {
            $query->join(
                ['ici' => 'icinga_command_inheritance'],
                'ici.command_id = c.id',
                []
            )->where('ici.parent_command_id = ?', $id);
        }

        foreach ($rels as $rel) {
            $query->orWhere("{$rel}_id = ?", $id);
        }
```